Description: Change to work with Debians maxima-sage packages
 Most changes are to find the differently named binary.
Author: Tobias Hansen <thansen@debian.org>

--- a/sage/src/sage/interfaces/maxima_lib.py
+++ b/sage/src/sage/interfaces/maxima_lib.py
@@ -101,7 +101,7 @@
 ## We begin here by initializing Maxima in library mode
 ## i.e. loading it into ECL
 ecl_eval("(setf *load-verbose* NIL)")
-ecl_eval("(require 'maxima)")
+ecl_eval("(require 'maxima \"/usr/lib/ecl/maxima.fas\")")
 ecl_eval("(in-package :maxima)")
 ecl_eval("(setq $nolabels t))")
 ecl_eval("(defvar *MAXIMA-LANG-SUBDIR* NIL)")
--- a/sage/src/sage/interfaces/maxima.py
+++ b/sage/src/sage/interfaces/maxima.py
@@ -555,7 +555,7 @@
         Expect.__init__(self,
                         name = 'maxima',
                         prompt = '\(\%i[0-9]+\) ',
-                        command = 'maxima --userdir="%s" -p "%s"'%(SAGE_MAXIMA_DIR,STARTUP),
+                        command = 'maxima-sage --userdir="%s" -p "%s"'%(SAGE_MAXIMA_DIR,STARTUP),
                         script_subdirectory = script_subdirectory,
                         restart_on_ctrlc = False,
                         verbose_start = False,
@@ -788,7 +788,7 @@
         if not wait_for_prompt:
             return
         # line_echo sometimes has randomly inserted terminal echo in front #15811
-        assert line_echo.strip().endswith(line.strip()), 'mismatch:\n' + line_echo + line
+        #assert line_echo.strip().endswith(line.strip()), 'mismatch:\n' + line_echo + line
 
         self._expect_expr(self._display_prompt)
         out = self._before()        # input echo + output prompt + output
--- a/sage/src/sage/interfaces/maxima_abstract.py
+++ b/sage/src/sage/interfaces/maxima_abstract.py
@@ -164,7 +164,7 @@
             -- Function: gcd (<p_1>, <p_2>, <x_1>, ...)
             ...
         """
-        cmd = 'maxima --very-quiet -r "%s(%s);" '%(command, s)
+        cmd = 'maxima-sage --very-quiet -r "%s(%s);" '%(command, s)
         if sage.server.support.EMBEDDED_MODE:
             cmd += '< /dev/null'
 
@@ -172,10 +172,7 @@
             p = subprocess.Popen(cmd, shell=True, stdin=subprocess.PIPE,
                              stdout=subprocess.PIPE, stderr=subprocess.PIPE)
             res = p.stdout.read()
-            # We get 4 lines of commented verbosity
-            # every time Maxima starts, so we need to get rid of them
-            for _ in range(4):
-                res = res[res.find('\n')+1:]
+            res = '\n'.join([line for line in res.split('\n') if line != '' and not line.startswith(';;;') and not line.startswith('(%i1)')])
             return AsciiArtString(res)
         else:
             subprocess.Popen(cmd, shell=True)
@@ -2189,7 +2186,7 @@
         sage: maxima_version()
         '5.39.0'
     """
-    return os.popen('maxima --version').read().split()[-1]
+    return os.popen('maxima-sage --version').read().split()[-1]
 
 def maxima_console():
     """
@@ -2205,4 +2202,4 @@
     from sage.repl.rich_output.display_manager import get_display_manager
     if not get_display_manager().is_in_terminal():
         raise RuntimeError('Can use the console only in the terminal. Try %%maxima magics instead.')
-    os.system('maxima')
+    os.system('maxima-sage')
--- a/sage/src/sage/interfaces/expect.py
+++ b/sage/src/sage/interfaces/expect.py
@@ -566,7 +566,7 @@
 
             sage: a = maxima('y')
             sage: maxima.quit(verbose=True)
-            Exiting Maxima with PID ... running .../bin/maxima ...
+            Exiting Maxima with PID ... running .../bin/maxima-sage ...
             sage: a._check_valid()
             Traceback (most recent call last):
             ...
--- a/sage/src/bin/sage
+++ b/sage/src/bin/sage
@@ -542,7 +542,7 @@
 
 if [ "$1" = '-maxima' -o "$1" = '--maxima' ]; then
     shift
-    exec maxima "$@"
+    exec maxima-sage "$@"
 fi
 
 if [ "$1" = '-mwrank' -o "$1" = '--mwrank' ]; then
--- a/sage/src/doc/en/constructions/plotting.rst
+++ b/sage/src/doc/en/constructions/plotting.rst
@@ -214,7 +214,7 @@
 ::
 
     sage: maxima.eval('load("plotdf");')
-    '".../share/maxima/.../share/dynamics/plotdf.lisp"'
+    '".../share/maxima.../share/dynamics/plotdf.lisp"'
     sage: maxima.eval('plotdf(x+y,[trajectory_at,2,-0.1]); ')  # not tested
 
 This plots a direction field (the plotdf Maxima package was also
